core/server/lib/members/index.js

```diff
@@ -7,20 +9,43 @@ module.exports = function MembersApi() {
     const apiRouter = Router();
 
     apiRouter.post('/token', (req, res) => {
+        const {signedin} = cookie.parse(req.headers.cookie);
+        if (!signedin) {
+            res.writeHead(401);
+            return res.end();
+        }
         const token = jwt.sign({}, null, {algorithm: 'none'});
         return res.end(token);
     });
 
-    apiRouter.post('/signin', (req, res) => {
+    apiRouter.post('/signin', body.json(), (req, res) => {
```

Comment:
I think its better extract out json body-parser method here to common router middleware on top as we'll be only sending json body and then it can be re-used for other routes 😄 
``` apiRouter.use(bodyParser.json())```

Reply:
If we do that then this would no longer be independent of ghost - so if we were to split out it would have a prerequisite of installing and setting up body parsing middleware!

Reply:
🤔 By common router middleware I meant on top of this file after we setup `apiRouter`. We are already using `body-parser` inside this particular post request here, so just moving to top on this file shouldn't need anything different right?